src/table/header-cell/th-element.tsx

```diff
@@ -0,0 +1,69 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import clsx from 'clsx';
+import React from 'react';
+import { TableProps } from '../interfaces';
+import { getAriaSort, getSortingStatus } from './utils';
+import styles from '../body-cell/styles.css.js';
+import rootStyles from '../styles.css.js';
+import { useVisualRefresh } from '../../internal/hooks/use-visual-mode';
+
+export interface TableThElementProps {
+  style?: React.CSSProperties;
+  sortable?: boolean;
+  sorted?: boolean;
+  sortingDescending?: boolean;
+  sortingDisabled?: boolean;
+  resizableColumns?: boolean;
+  hidden?: boolean;
+  sticky?: boolean;
+  stuck?: boolean;
+  stripedRows?: boolean;
+  hasSelection?: boolean;
```

Comment:
Are these all optional?

Optional props lead to unused code, as it was demonstrated in the past: https://github.com/cloudscape-design/components/commit/368b042255e4fc257a2dab72b25c38818a16663c

If `undefined` is supported value, pass it as `| undefined`